cardano-api/gen/Gen/Cardano/Api/Typed.hs

```diff
@@ -667,50 +674,50 @@ genProtocolParameters =
-    <*> Gen.maybe genExecutionUnits
-    <*> Gen.maybe genExecutionUnits
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-
-genProtocolParametersUpdate :: Gen ProtocolParametersUpdate
-genProtocolParametersUpdate =
-  ProtocolParametersUpdate
-    <$> Gen.maybe ((,) <$> genNat <*> genNat)
-    <*> Gen.maybe genRational
-    <*> Gen.maybe genMaybePraosNonce
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genLovelace
-    <*> Gen.maybe genLovelace
-    <*> Gen.maybe genLovelace
-    <*> Gen.maybe genLovelace
-    <*> Gen.maybe genEpochNo
-    <*> Gen.maybe genNat
-    <*> Gen.maybe genRationalInt64
-    <*> Gen.maybe genRational
-    <*> Gen.maybe genRational
-    <*> Gen.maybe genLovelace
-    <*> genCostModels
+    <*> genCostModels era
     <*> Gen.maybe genExecutionUnitPrices
     <*> Gen.maybe genExecutionUnits
     <*> Gen.maybe genExecutionUnits
     <*> Gen.maybe genNat
     <*> Gen.maybe genNat
     <*> Gen.maybe genNat
 
-
-genUpdateProposal :: Gen UpdateProposal
-genUpdateProposal =
+genProtocolParametersUpdate :: CardanoEra era -> Gen ProtocolParametersUpdate
+genProtocolParametersUpdate era = do
+  protocolUpdateProtocolVersion     <- Gen.maybe ((,) <$> genNat <*> genNat)
+  protocolUpdateDecentralization    <- Gen.maybe genRational
+  protocolUpdateExtraPraosEntropy   <- Gen.maybe genMaybePraosNonce
+  protocolUpdateMaxBlockHeaderSize  <- Gen.maybe genNat
+  protocolUpdateMaxBlockBodySize    <- Gen.maybe genNat
+  protocolUpdateMaxTxSize           <- Gen.maybe genNat
+  protocolUpdateTxFeeFixed          <- Gen.maybe genNat
+  protocolUpdateTxFeePerByte        <- Gen.maybe genNat
+  protocolUpdateMinUTxOValue        <- untilAlonzo era genLovelace
+  protocolUpdateStakeAddressDeposit <- Gen.maybe genLovelace
+  protocolUpdateStakePoolDeposit    <- Gen.maybe genLovelace
+  protocolUpdateMinPoolCost         <- Gen.maybe genLovelace
+  protocolUpdatePoolRetireMaxEpoch  <- Gen.maybe genEpochNo
+  protocolUpdateStakePoolTargetNum  <- Gen.maybe genNat
+  protocolUpdatePoolPledgeInfluence <- Gen.maybe genRationalInt64
+  protocolUpdateMonetaryExpansion   <- Gen.maybe genRational
+  protocolUpdateTreasuryCut         <- Gen.maybe genRational
+  protocolUpdateUTxOCostPerWord     <- ifAlonzoBased era genLovelace
+  protocolUpdateCostModels          <- genCostModels era
+  protocolUpdatePrices              <- ifAlonzoBased era genExecutionUnitPrices
+  protocolUpdateMaxTxExUnits        <- ifAlonzoBased era genExecutionUnits
+  protocolUpdateMaxBlockExUnits     <- ifAlonzoBased era genExecutionUnits
+  protocolUpdateMaxValueSize        <- ifAlonzoBased era genNat
+  protocolUpdateCollateralPercent   <- ifAlonzoBased era genNat
+  protocolUpdateMaxCollateralInputs <- ifAlonzoBased era genNat
+  pure ProtocolParametersUpdate{..}
+
+
+genUpdateProposal :: CardanoEra era -> Gen UpdateProposal
```

Comment:
For example, I doubt this needs to be era-dependent now. Isn't it just the ProtocolParameters where we have a new property that needs a value of the parameters with certain params set?

Reply:
genProtocolParametersUpdate has 9 era-dependent fields.

I'm not sure I fully understand the question.

Do you want to generate all parameters regardless of the era? And just ignore unknown parameters in the end?